Assets/Character/AaronTestAssets/PlayerState.cs

```diff
@@ -0,0 +1,74 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class PlayerState : MonoBehaviour {
+
+
+    private Vector3 scale;
+    protected float direction = 1;
+    protected int facing = 1;
+    protected bool grounded = false;
+    protected new string name;
+    protected SpriteRenderer sr;
+    protected IdleState idle;
+    protected RunState run;
+    protected RollState roll;
+    protected AttackState attack;
+    protected BackstepState backstep;
+    protected AirborneState airborne;
+
+    //this doesn't appear to do anything...
+    void Start ()
+    {
+        sr = gameObject.GetComponent<SpriteRenderer>();
+
+    }
+
+    // Update is called once per frame
+    public virtual void Update ()
+    {
+
+    }
+
+    public virtual void ComponentUpdate()
+    {
+
+    }
+
+    public virtual void handleInput()
+    {
+
+    }
+
+    public void Flip()
+    {
+
+
+        if (facing > 0 && direction < 0)
+        {
+            gameObject.GetComponent<SpriteRenderer>().flipX = true;
```

Comment:
This is the sr member variable right? If you need the sprite renderer everywhere use sr, otherwise remove it as a member and stick to this when needed.